topostats/tracing/tracingfuncs.py

```diff
@@ -531,3 +531,26 @@ def coord_dist(coords: npt.NDArray, px_2_nm: float = 1) -> npt.NDArray:
             dist += 1
         dist_list.append(dist)
     return np.asarray(dist_list) * px_2_nm
+
+
+def get_two_combinations(list_of_values) -> list:
+    """
+    Obtain all paired combinations of values in the list.
+
+    Example: [1,2] -> [[1,2]], [1,2,3] -> [[1,2],[1,3],[2,3]]
+
+    Parameters
+    ----------
+    list_of_values : list
+        List of numbers.
+
+    Returns
+    -------
+    list
+        A list of combinatorics pairs of values, retaining the initial order.
+    """
+    combs = []
+    for i in range(len(list_of_values) - 1):
```

Comment:
Probably faster to leverage [`itertools.combinations()`](https://docs.python.org/3/library/itertools.html#itertools.combinations).

Could then return

```python
def get_combinations(list_of_values, r)
return list(itertools.combinations(list_of_values, r))
```

Documentation states it retains the sort order of the input values.

This may mean the whole function is redundant and instead the calls to `get_two_combinations()` could be replaced with `combinations(<some_list>, <size)`.

Reply:
Cool! I've subbed that in but it also needed converting to a list